lib/mfx/api.py

```diff
@@ -0,0 +1,373 @@
+        51  : MFX_LEVEL_AV1_51,    
+        52  : MFX_LEVEL_AV1_52,    
+        60  : MFX_LEVEL_AV1_6,    
+        61  : MFX_LEVEL_AV1_61,    
+        62  : MFX_LEVEL_AV1_62,    
+      },
+      #/* HEVC levels */
+      Codec.HEVC   : {
+        10  : MFX_LEVEL_HEVC_1,    
+        20  : MFX_LEVEL_HEVC_2,    
+        21  : MFX_LEVEL_HEVC_21,    
+        30  : MFX_LEVEL_HEVC_3,    
+        31  : MFX_LEVEL_HEVC_31,    
+        40  : MFX_LEVEL_HEVC_4,    
+        41  : MFX_LEVEL_HEVC_41,    
+        50  : MFX_LEVEL_HEVC_5,    
+        51  : MFX_LEVEL_HEVC_51,    
+        52  : MFX_LEVEL_HEVC_52,    
+        60  : MFX_LEVEL_HEVC_6,    
+        61  : MFX_LEVEL_HEVC_61,    
+        62  : MFX_LEVEL_HEVC_62,    
+      },
+      #/* AV1 Levels */
+      Codec.AV1 : {
+        20  : MFX_LEVEL_AV1_2,    
+        21  : MFX_LEVEL_AV1_21,    
+        22  : MFX_LEVEL_AV1_22,    
+        23  : MFX_LEVEL_AV1_23,    
+        30  : MFX_LEVEL_AV1_3,    
+        31  : MFX_LEVEL_AV1_31,    
+        32  : MFX_LEVEL_AV1_32,    
+        33  : MFX_LEVEL_AV1_33,    
+        40  : MFX_LEVEL_AV1_4,    
+        41  : MFX_LEVEL_AV1_41,    
+        42  : MFX_LEVEL_AV1_42,    
+        43  : MFX_LEVEL_AV1_43,    
+        50  : MFX_LEVEL_AV1_5,    
+        51  : MFX_LEVEL_AV1_51,    
+        52  : MFX_LEVEL_AV1_52,    
+        53  : MFX_LEVEL_AV1_53,    
+        60  : MFX_LEVEL_AV1_6,    
+        61  : MFX_LEVEL_AV1_61,    
+        62  : MFX_LEVEL_AV1_62,    
+        63  : MFX_LEVEL_AV1_63,    
+        70  : MFX_LEVEL_AV1_7,    
+        71  : MFX_LEVEL_AV1_71,    
+        72  : MFX_LEVEL_AV1_72,    
+        73  : MFX_LEVEL_AV1_73,    
+      },
+    }[codec][level]
+
+  @classmethod
+  @memoize
+  def map_rcmode(cls, rcmode):
+    # TODO: we could move this method to the enum class as "lookup"
+    # then call MfxRateControlMethod.lookup(rcmode)
+    return {
+      "CQP" : RateControlMethod.MFX_RATECONTROL_CQP,
+      "CBR" : RateControlMethod.MFX_RATECONTROL_CBR,
+      "VBR" : RateControlMethod.MFX_RATECONTROL_VBR,
+      "ICQ" : RateControlMethod.MFX_RATECONTROL_ICQ,
+    }[rcmode.upper()]
+
+# Example:
+#
+# <file lib/ffmpeg/qsv/encoder.py>
+#
+# # NOTE: Inherit from the StringAPI class first so that it overrides the
+# # Encoder class
+# from ....lib.mfx.api import StringAPI
+# class StringAPIEncoder(StringAPI, Encoder):
+#   @property
+#   def encparams(self):
+#     return f" -qsv_params '{super().encparams}'"
+#
+# <file test/ffmpeg-qsv/encode/hevc.py>
+#
+# from ....lib.ffmpeg.qsv.encoder import StringAPIEncoder
+# class cqp(HEVC8EncoderTest):
+#   ...
+#   @slash.requires(*have_string_api) # TODO: define have_string_api
+#   @slash.parametrize(*gen_hevc_cqp_parameters(spec, ['main']))
+#   def test_strapi(self, case, gop, slices, bframes, qp, quality, profile):
+#     self.EncoderClass = StringAPIEncoder
+#     self.init(spec, case, gop, slices, bframes, qp, quality, profile)
+#     self.encode()
+#
+#
```

Comment:
This comment block can be removed.  It is stale.

The actual implementations provided by your PR can be used as a future reference, instead.